src/providers/npm.rs

```diff
@@ -42,7 +42,7 @@ impl Provider for NpmProvider {
     }
 
     fn install(&self, _app: &App, _env: &Environment) -> Result<Option<InstallPhase>> {
-        let install_phase = InstallPhase::new("npm ci".to_string());
+        let install_phase = InstallPhase::new("npm ci || npm install".to_string());
```

Comment:
I think checking if `package-lock.json` exists and then running `npm ci` if it does is a better idea.

Reply:
If dependencies in the package lock do not match those in package.json npm ci fails which isn't ideal. Also, it's not just `package-lock.json` ci works with `npm-shrinkwrap.json` too

Reply:
I agree with Milo. The entire point of the lockfile is to set exact versions of a dependency. If the versions in the lockfile do not match those in package.json then building with Nixpacks **should** fail.

However, if you don't have a lockfile specified then that is okay and we can fallback to just running `npm install`.

Reply:
Buildpacks don't do that and I have seen repos (including  [railway starters](https://github.com/railwayapp-starters/ghost/commit/9b31609b8fb503657716a550d4d51dc11aef8bf4)) where people just update package.json temporarily to update to a new version of that dependency. Won't be backward compatible if we make this change. Up to you/team I don't mind adding that.

Reply:
I think the Paketo buildpacks also install with `npm ci` or `yarn --frozen-lockfile`. IMO when updating a version the lockfile should also be regenerated.

In this case I think we should respect the lockfile contents.

Reply:
Sure, will update PR accordingly.